Sources/Sentry/SentryThreadInspector.m

```diff
@@ -12,6 +18,36 @@
 
 @end
 
+typedef struct {
+    SentryCrashThread thread;
+    SentryCrashStackEntry stackEntries[MAX_STACKTRACE_LENGTH];
+    int stackLength;
+} ThreadInfo;
+
+// We need a C function to retrieve information from the stack trace in order to avoid
+// calling into async code while there is suspended threads.
+unsigned int
+getStackEntriesFromThread(SentryCrashThread thread, struct SentryCrashMachineContext *context,
+    SentryCrashStackEntry *buffer, unsigned int amount)
+{
+    sentrycrashmc_getContextForThread(thread, context, false);
+    SentryCrashStackCursor stackCursor;
+    sentrycrashsc_initWithMachineContext(&stackCursor, MAX_STACKTRACE_LENGTH, context);
+
+    unsigned int result = 0;
+    while (stackCursor.advanceCursor(&stackCursor)) {
+        if (result == amount)
+            break;
+        if (stackCursor.symbolicate(&stackCursor)) {
+            buffer[result] = stackCursor.stackEntry;
+            result++;
+        }
+    }
+    sentrycrash_async_backtrace_decref(stackCursor.async_caller);
```

Comment:
`m`: I think we should have one test with `options.stitchAsyncCode = true` and check if the code doesn't blow up.